src/personalizedFeed.ts

```diff
@@ -23,7 +24,8 @@ async function fetchTinybirdFeed(
     url += `&user_id=${userId}`;
   }
   if (feedId) {
-    const filters = await feedToFilters(con, feedId);
+    const features = await fetchUserFeatures(feedId);
+    const filters = await feedToFilters(con, feedId, features);
```

Comment:
I think there's a way to run these two in parallel if we do the merge between them on the server-side instead of the database side as you do now.
So from the database, we can fetch the defaults and the user settings and if the advanced settings defaults are enabled we can override. It will help us cut a few ms probably.
What do you think?

Reply:
Another suggestion would be to add the fetching of the flags into the feed filters function.
This will make sure that whenever we use this function we get the same results and keep it consistent.

Reply:
Hmm, that's a very interesting idea. Though for us to merge, we will have to again run the same query for pulling all the sources to exclude. I think the Query will have to be called twice but I agree that it will be able to run in parallel which is also very fast and not wait for anything. Really interesting.

Initially, I applied the fetching of features to the feed filters but I realized our anonymous feed goes through here as well. It became inapplicable for some cases and just had to inject only if necessary.

Reply:
Oh you are right! Then forget what I mentioned about making it parallel.

As for injecting the parameters, I see what you mean. So please make sure this parameter is a required parameter and not optional so we will inject it everywhere.
Because I didn't see you updated the GraphQL query that returns the defaults for the advanced settings

Reply:
But it should be optional right? Since anonymous feed doesn't have a user id and it is the user id that we use to fetch our features (flagsmith) whether it is enabled for their account or not.

Reply:
We have a user id for every user even the not logged-in ones. We should use it for the anonymous feed as well.

Reply:
Oh, cool. Then we would really have to make it necessary and be inside the feedToFilters function since this will be now used across all cases.